src/impl_pipeline.cpp

```diff
@@ -386,6 +386,385 @@ auto daxa_compute_pipeline_dec_refcnt(daxa_ComputePipeline self) -> u64
+
+        auto& raygen_region = ret.info.shader_binding_table.raygen_region;
+        auto& miss_region = ret.info.shader_binding_table.miss_region;
+        auto& hit_region = ret.info.shader_binding_table.hit_region;
+        auto& callable_region = ret.info.shader_binding_table.callable_region;
+
+        raygen_region.stride = get_aligned(handle_size_aligned, handle_stride);
+        raygen_region.size = raygen_region.stride;  // The size member of pRayGenShaderBindingTable must be equal to its stride member
+        miss_region.stride = handle_size_aligned;
+        miss_region.size = get_aligned(miss_count_number * handle_size_aligned, handle_stride);
+        hit_region.stride = handle_size_aligned;
+        hit_region.size = get_aligned(hit_count_number * handle_size_aligned, handle_stride);
+        callable_region.stride = handle_size_aligned;
+        callable_region.size = get_aligned(callable_count_number * handle_size_aligned, handle_stride);
+
+        // Get the shader group handles
+        u32 data_size = handle_count * handle_size;
+        std::vector<uint8_t> shader_handle_storage(data_size);
+        // Allocate a buffer for storing the SBT.
+        VkDeviceSize sbt_size = raygen_region.size + miss_region.size + hit_region.size + callable_region.size;
+        // https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetRayTracingShaderGroupHandlesNV.html
+        auto shader_group_result = ret.device->vkGetRayTracingShaderGroupHandlesKHR(
+            ret.device->vk_device, 
+            ret.vk_pipeline,
+            0,
+            handle_count,
+            sbt_size,
+            shader_handle_storage.data());
+
+        if (shader_group_result == VK_SUCCESS) {
+
+            auto name_cstr = ret.info.name.c_str();
+            // Allocate a buffer for storing the SBT.
+            auto stb_info = daxa_BufferInfo{
+                .size = sbt_size,
+                .allocate_info = DAXA_MEMORY_FLAG_HOST_ACCESS_SEQUENTIAL_WRITE,
+                .name = std::bit_cast<daxa_SmallString>(name_cstr),
+            };
+            // TODO: We need to store the buffer id somewhere, so we can destroy after the pipeline is destroyed
+            auto& stb_buffer_id = ret.info.shader_binding_table.buffer_id;
+            auto buffer_result = daxa_dvc_create_buffer(device, &stb_info, r_cast<daxa_BufferId *>(&stb_buffer_id));
+            if (buffer_result == DAXA_RESULT_SUCCESS)
+            {
+
+                u8* sbt_buffer_ptr;
+                auto host_buffer_address_result = daxa_dvc_buffer_host_address(device, stb_buffer_id, reinterpret_cast<void**>(&sbt_buffer_ptr));
+                if(host_buffer_address_result != DAXA_RESULT_SUCCESS)
+                {
+                    // TODO: Check this
+                    pipeline_result = VK_ERROR_OUT_OF_HOST_MEMORY;
+                    auto destroy_buffer_result = daxa_dvc_destroy_buffer(device, stb_buffer_id);
+                    goto cleanup;
```

Comment:
See if Patrick has gripes regarding goto

Reply:
I know that goto is a polemic statement but I like it when you want to avoid a lot of conditionals.

Reply:
We have a pattern for this, which we actually have not put in place (and should) which is `defer` as defined in `impl_core.hpp`. you can put at the top of the function 
```cpp
defer {
  // clean-up code
}
```
and so all goto calls just get turned into `return ...`